hoodie-common/src/main/java/com/uber/hoodie/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -822,10 +832,18 @@ public void sync() {
    * @param newTimeline New Hoodie Timeline
    */
   protected void runSync(HoodieTimeline oldTimeline, HoodieTimeline newTimeline) {
-    visibleActiveTimeline = newTimeline;
+    updateTimeline(newTimeline);
     addedPartitions.clear();
     resetViewState();
     // Initialize with new Hoodie timeline.
     init(metaClient, newTimeline);
   }
+
+  /**
+   * Return Only Commits and Compaction timeline for building file-groups
+   * @return
+   */
+  public HoodieTimeline getVisibleCommitsAndCompactionTimeline() {
```

Comment:
is this needed?  seems redundant with `getTimeline()`?

Reply:
This is needed to distinguish active timeline from commitsAndCompactionTimeline